File: chapter-4/java/demo-web/src/main/java/my/app/demo/Employee.java
package my.app.demo;

import com.redis.om.spring.annotations.Document;
import com.redis.om.spring.annotations.Indexed;
import com.redis.om.spring.annotations.Searchable;
import lombok.*;
import org.springframework.data.annotation.Id;

import java.util.Set;

@RequiredArgsConstructor(staticName = "of")
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Data
@Document
public class Employee {

    @Id
    @Indexed
    private String id;

    @Indexed @NonNull
    private String firstName;

    @Indexed @NonNull
    private String lastName;

    @Indexed @NonNull
    private Integer age;

    @Searchable @NonNull
    private String funFact;

    @Indexed @NonNull
    private Office office;

    @Indexed @NonNull
    private Set<String> roles;
}

File: chapter-4/dotnetcore/dotnetcsharp/Employee.cs
using Redis.OM.Modeling;

namespace dotnetcsharp;

[Document(StorageType = StorageType.Json, Prefixes = new []{"Employee"})]
public class Employee
{
    [RedisIdField] [Indexed] public string? Id { get; set; }
    [Indexed] public string? FirstName { get; set; }
    [Indexed] public string? LastName { get; set; }
    [Indexed] public int? Age { get; set; }
    [Searchable] public string? FunFact { get; set; }
    [Indexed(CascadeDepth = 1)] public Office? Office { get; set; }
    [Indexed] public string[] Roles { get; set; } = Array.Empty<string>();
}